### DISCUSSION TITLE
Create sheetmetal from surface body

### DISCUSSION THREAD

**Discussion by rmdug** [shaise/FreeCAD_SheetMetal#380] [Ideas]
This proposal is linked to this non-issue : https://github.com/shaise/FreeCAD_SheetMetal/issues/378

In sheetmetal design it can be very handy to create first a surface body and then turn it to a sheetmetal:
+ surface creation
For example:
![image](https://github.com/user-attachments/assets/10ee7647-31f7-4e7b-812b-877fc66db574)

+ sheetmetal creation:
    + set thickness;
    + set bend data (radius, K factor,…);
    + set corner data and free edges joints,

+ selection of bent edges
For example (bent edges in red):
![image](https://github.com/user-attachments/assets/55fb59b3-f54c-4e98-aa5e-8164e8c512ae)

This way of doing proved to be very efficient to do tricky shapes including unusual bend angles with the commercial software I use.
Thank you for your great work : )

**Comment by luzpaz** [comment]
@shaise can you weigh-in ?

**Reply by shaise** [reply]
Yes, this issue was already discussed. This is a nice feature to have though quite complicated to implement.

**Reply by luzpaz** [reply]
@shaise is there a FR ticket for it ?

**Reply by shaise** [reply]
Sorry, what do you mean by FR?

**Reply by luzpaz** [reply]
Feature Request

**Reply by shaise** [reply]
Ah, dont think so. I thought there is a label for it as an Issue, but did not find one.

**Reply by luzpaz** [reply]
Created a ticket from this discussion: https://github.com/shaise/FreeCAD_SheetMetal/issues/385

**Reply by PeterPNoster** [reply]
I would like to support such a feature.
Based on the wish to have a weldable edge. From discussion: [https://forum.freecad.org/viewtopic.php?t=93290](url)

**Reply by shaise** [reply]
This is not simple to implement but definitely on my list.

**Comment by luzpaz** [comment]
Closing in favor of #385
